### PR TITLE
fix(frontend): Show recurring run display name rather than ID. Fix #1786

### LINKED ISSUE #1786: Show Recurring run name instead of id in run page
**What happened:**
image

**What did you expect to happen:**
Show recurring run name

**What steps did you take:**
[A clear and concise description of what the bug is.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by zijianjoy**
**Description of your changes:**
Fix #1786 : We are able to retrieve the job name in single API call, thus we can show display name rather than job ID.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: frontend/src/pages/RunList.test.tsx [hunk 1]
```diff
@@ -434,7 +439,75 @@ describe('RunList', () => {
     tree = shallow(<RunList {...props} />);
     await (tree.instance() as RunListTest)._loadRuns({});
     expect(props.onError).not.toHaveBeenCalled();
-    expect(tree).toMatchSnapshot();
+    expect(tree).toMatchInlineSnapshot(`
```
Comment (StefanoFioravanzo) [inline frontend/src/pages/RunList.test.tsx:442]: Why did you change this to be an inline snapshot?

**Review by StefanoFioravanzo — COMMENTED** [review]
This is great!

File: frontend/src/pages/RunList.test.tsx (hunk 1 above)
Comment (zijianjoy) [reply frontend/src/pages/RunList.test.tsx:442]: I am following the testing best practices in https://github.com/kubeflow/pipelines/pull/3166, which allows us to inspect the UI element outcome on the spot. But feel free to argue differently. :)

**Comment by Bobgy** [comment]
/lgtm
/approve
Awesome! I think it's ready to go
